### PR TITLE
Fix: Hide nonce input if its a spending limit transaction

### LINKED ISSUE #2880: Spending limits - The nonce field is visible when non-owners use the allowance
## Bug description
Non-owners can start a send funds tx if they have an allowance set in the spending limit feature. The issue is that the Nonce field is present in the form.
There are no issues with creating/executing the tx itself. Is just the nonce field that should not be visible

## Environment
 - Browser: Chrome
 - Wallet: MetaMask
 - Chain: Any
 - Env: Prod v1.23

## Steps to reproduce
 1. Go to a safe you own
 2. Create a spending limit of any token for a non-owner wallet you control
 3. Connect with that wallet and go into that safe
 4. Start a Send funds tx
 5. The spending limit option should be selected by default in the form
 6. Check the nonce input

## Expected result
The nonce input should not be visible

## Obtained result
The nonce input is showing. I'd assume that the form hides it when you select "use spending limit" in the form, but since is the default option you never click it, therefore the form never checks

## Screenshots
image

### PR DESCRIPTION
**Pull request by usame-algan**
## What it solves

Resolves #2880 

## How this PR fixes it

- Hides the nonce input field if a non-owner tries to create a spending limit transaction
- Pre-selects the first available token when a non-owner tries to create a spending limit transaction
- Displays the nonce input again if an owner switches from a spending limit token to a non-spending limit token

## How to test it

1. Open a safe with a spending limit as a non-owner
2. Create a transaction and observe the nonce input is hidden
3. Observe a token is pre-selected in the form

---

1. Open a safe as an owner that also has a spending limit
2. Create a transaction
3. Observe the nonce input being visible
5. Select the spending limit token and select spending limit in the radio group
6. Observe the nonce input is hidden
7. Switch to a different token
8. Observe the nonce input is visible

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Comment by usame-algan** [comment]
I tried to keep the changes as minimal as possible. The third case is another bug I found that wasn't reported yet. I want to write tests for this component but would also refactor it in the process because its overly complex at the moment. Wdyt about doing that in this PR?

File: src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx [hunk 1]
```diff
@@ -67,6 +67,16 @@ const CreateTokenTransfer = ({
...
 
+  const balancesItems = useMemo(() => {
+    return isOnlySpendingLimitBeneficiary
+      ? balances.items.filter(({ tokenInfo }) => {
+          return spendingLimits?.some(({ beneficiary, token }) => {
+            return sameAddress(beneficiary, wallet?.address || '') && sameAddress(tokenInfo.address, token.address)
+          })
+        })
+      : balances.items
+  }, [balances.items, isOnlySpendingLimitBeneficiary, spendingLimits, wallet?.address])
```
Comment (usame-algan) [inline src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx:78]: This is the same as it was before just pushed further up so we can use it to set the default form value.

File: src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx [hunk 2]
```diff
@@ -133,6 +134,12 @@ const CreateTokenTransfer = ({
   const isDisabled = isSafeTokenSelected && isSafeTokenPaused
   const isAddressValid = !!recipient && !errors[TokenTransferFields.recipient]
 
+  useEffect(() => {
+    if (!spendingLimit) {
+      setNonceNeeded(true)
+    }
```
Comment (katspaugh) [inline src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx:140]: Will it flip back if they switch to a spending limit?

Comment (schmanu) [reply src/components/tx-flow/flows/TokenTransfer/CreateTokenTransfer.tsx:140]: Yes the `SpendingLimitRow` `onChange` handler also sets setNonceNeeded.

**Review by schmanu — APPROVED** [review]
At some point we should write test for the CreateTokenTransfer component.
We should create a tech debt issue.

**Comment by francovenica** [comment]
Issue:

Gif:
This is an owner I'm connected with
He has spending limit allowance for tokenA, but not for tokenB
He selects tokenA, selects spending limit and the nonce goes away (expected)
When switching to tokenB the nonce shows up again, since he can only do a regular tx, (expected as well)
When switching back to tokenA the spending limit option is marked by default and the nonce input keeps showing (not expected)

Is expected that the nonce field is hidden once again if the "Spending limit" option is selected
spending limit

**Comment by usame-algan** [comment]
> Is expected that the nonce field is hidden once again if the "Spending limit" option is selected

Thanks for checking! I updated it so the nonce input is hidden again.

**Comment by francovenica** [comment]
Issue fixed. 

The rest also looks good to me.

**Review by schmanu — APPROVED** [review]

**Pull request merged**